Copy and adjust vivarium modeling strategy from sepsis to OL page (#1553)
ihmeuw/vivarium_research: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst

Scope
+++++

The goal of the obstructed labor (OL) model is to capture YLLs and YLDs due to
obstructed labor and uterine rupture among women of
reproductive age. We only model obstructed labor and uterine rupture among 
simulants who give (live or still) birth after a full term pregnancy. This 
page documents how to model the baseline burden of obstructed labor and 
uterine rupture. Other simulation components such as c-sections will affect 

> the rates of obstructed labor; the pages for the corresponding :ref:`intervention <intervention_models>`

Reviewer: ```suggestion
the rates of obstructed labor; such effects will be described on the pages for the corresponding :ref:`intervention <intervention_models>`
```

cases to calculate YLDs.

Assumptions and Limitations
+++++++++++++++++++++++++++

Cause Model Decision Graph
++++++++++++++++++++++++++

Although we're not modeling obstructed labor and uterine rupture
dynamically as a finite state machine, we can draw an analogous directed 
graph that can be interpreted as a (collapsed) decision tree rather than 
a state transition diagram. The main difference is that the values on the 
transition arrows represent decision probabilities rather than rates per 
unit time. The obstructed labor and uterine rupture decision graph drawn 
below should be inserted on the "full term pregnancy" branch of the decision 
graph from the :ref:`pregnancy model <other_models_pregnancy_closed_cohort_mncnh>`,
between the intrapartum model and the birth of the child simulant. Solid
lines are the pieces added by the obstructed labor and uterine rupture model, 
while dashed lines indicate pieces of the underlying pregnancy model.

.. graphviz::

    digraph OL_decisions {
        rankdir = LR;
        ftp [label="full term\npregnancy, post\nintrapartum", style=dashed]
        ftb [label="full term\nbirth", style=dashed]

>         alive [label="parent did not die of OL"]

Reviewer: ```suggestion
        alive [label="parent did not\ndie of OL"]
```
Let's try splitting it into two lines to make the diagram less wide

>         dead [label="parent died of OL"]

Reviewer: ```suggestion
        dead [label="parent died\nof OL"]
```
ditto

        ftp -> alive  [label = "1 - ir"]
        ftp -> OL [label = "ir"]
        OL -> alive [label = "1 - cfr"]
        OL -> dead [label = "cfr"]
        alive -> ftb  [label = "1", style=dashed]
        dead -> ftb  [label = "1", style=dashed]
    }

.. list-table:: State Definitions
    :widths: 7 20
    :header-rows: 1

    * - State
      - Definition
    * - full term pregnancy, post intrapartum
      - Parent simulant has a full term pregnancy as determined by the
        :ref:`pregnancy model
        <other_models_pregnancy_closed_cohort_mncnh>`, **and** has
        already been through the antenatal and intrapartum models
    * - OL
      - Parent simulant experiences obstructed labor or uterine rupture

>     * - parent not dead of OL 

Author: @NathanielBlairStahn I updated these state names and definitions but not sure if the names are good!

Reviewer: ```suggestion
    * - parent did not die of OL 
```
Looks good! But I think we should match the wording in the diagram exactly.

      - Parent simulant did not die of obstructed labor or uterine rupture
    * - parent died of OL
      - Parent simulant died of obstructed labor or uterine rupture
    * - full term birth
      - The parent simulant has given birth to a child simulant (which
        may be a live birth or a still birth, to be determined in the
        next step of the :ref:`pregnancy model
        <other_models_pregnancy_closed_cohort_mncnh>`)

.. list-table:: Transition Probability Definitions
    :widths: 1 5 20
    :header-rows: 1

    * - Symbol
      - Name
      - Definition
    * - ir
      - incidence risk
      - The probability that a pregnant simulant experiences obstructed 
        labor or uterine rupture
    * - cfr
      - case fatality rate

>       - The probability that a simulant with  experiences dies of that 

Reviewer: ```suggestion
      - The probability that a simulant who experiences obstructed labor or uterine rupture dies of that 
```

Scope
+++++

The goal of the obstructed labor model is to capture YLLs and YLDs due to
obstructed labor and uterine rupture among women of
reproductive age. We only model obstructed labor and uterine rupture among 
simulants who give (live or still) birth after a full term pregnancy. This 
page documents how to model the baseline burden of obstructed labor and 
uterine rupture. Other simulation components such as c-sections will affect 

> the rates of the pages for the corresponding :ref:`intervention <intervention_models>`

Reviewer: ```suggestion
the rates of obstructed labor; the pages for the corresponding :ref:`intervention <intervention_models>`
```

cases to calculate YLDs.

Assumptions and Limitations
+++++++++++++++++++++++++++

Cause Model Decision Graph
++++++++++++++++++++++++++

Although we're not modeling obstructed labor and uterine rupture
dynamically as a finite state machine, we can draw an analogous directed 
graph that can be interpreted as a (collapsed) decision tree rather than 
a state transition diagram. The main difference is that the values on the 
transition arrows represent decision probabilities rather than rates per 
unit time. The obstructed labor and uterine rupture decision graph drawn 
below should be inserted on the "full term pregnancy" branch of the decision 
graph from the :ref:`pregnancy model <other_models_pregnancy_closed_cohort_mncnh>`,
between the intrapartum model and the birth of the child simulant. Solid
lines are the pieces added by the obstructed labor and uterine rupture model, 
while dashed lines indicate pieces of the underlying pregnancy model.

.. graphviz::

    digraph OL_decisions {
        rankdir = LR;
        ftp [label="full term\npregnancy, post\nintrapartum", style=dashed]
        ftb [label="full term\nbirth", style=dashed]
        alive [label="parent alive"]

>         dead [label="parent dead"]

Author: In PR [#1554](https://github.com/ihmeuw/vivarium_research/pull/1554#issue-2680986294) I changed these labels to 'parent did not die of cause' and 'parent died of cause' to be more accurate to our simulation.

Reviewer: FYI @pletale: I see that you branched https://github.com/ihmeuw/vivarium_research/pull/1554 off of this branch, but generally it's easier to avoid merge conflicts if you always branch new PRs off the `main` branch and edit completely disjoint sections of the page in each PR. (Granted, that's not always possible, but in this case it is.)

For example, if you make an edit in this PR based on feedback, then you'll have to merge the new edits from this PR into PR 1554 before that PR can be merged, which can lead to conflicts that need to be resolved, potentially introducing new errors (I've seen this happen before, requiring a follow-up PR to correct the merge errors). I added section headers to each page so that each PR can correspond to content between two section headers, which will ensure that the content in different PRs is disjoint, as long as the headers themselves are not edited (this was James Collins's suggested strategy for editing the docs).

In other words, it would probably be easier to make all the changes to the "Vivarium Modeling Strategy" and "Cause Model Decision Graph" sections in this PR, and in the other PR, _only_ make changes to the "Data Tables" section. So I would recommend _closing_ https://github.com/ihmeuw/vivarium_research/pull/1554, and moving the edits to the to the graph and to the "Scope" section to this PR instead, and copy all the edits to the "Data Tables" section into a _new_ PR that is branched off of `main`.

@zmbc @aflaxman @SylLutze Does this seem like generally good practice to you (i.e. keeping content in different PRs disjoint, and branching off `main` when possible)? Do any of you have any other suggestions for making things easy and avoiding merge conflicts?

Reviewer 2: I don't think the branching strategy has too much impact on the likelihood of merge conflicts -- as long as you have the section headers as you mention, you really shouldn't get conflicts. But generally I do think it is good to make separate PRs separate (not on top of each other) so they could be merged in any order.

Reviewer: > as long as you have the section headers as you mention, you really shouldn't get conflicts

In this case, https://github.com/ihmeuw/vivarium_research/pull/1554 is making edits to the sections in this PR, so I'm more confident that merge conflicts will be avoided if we simply create a new PR off of `main` after moving those edits to this PR, rather than try to revert those edits in the other PR.

> so they could be merged in any order

Ah, yes, that's another good reason to branch off `main` when possible, so that it doesn't get confusing when you try to merge everything. I've seen that happen, too, where merging in the wrong order required a follow-up PR to correct things: https://github.com/ihmeuw/vivarium_research/pull/1530

        ftp -> alive  [label = "1 - ir"]
        ftp -> OL [label = "ir"]
        OL -> alive [label = "1 - cfr"]
        OL -> dead [label = "cfr"]
        alive -> ftb  [label = "1", style=dashed]
        dead -> ftb  [label = "1", style=dashed]
    }

.. list-table:: State Definitions
    :widths: 7 20
    :header-rows: 1

    * - State
      - Definition
    * - full term pregnancy, post intrapartum
      - Parent simulant has a full term pregnancy as determined by the
        :ref:`pregnancy model
        <other_models_pregnancy_closed_cohort_mncnh>`, **and** has
        already been through the antenatal and intrapartum models
    * - OL
      - Parent simulant experiences obstructed labor or uterine rupture 

>         (fistula).

Reviewer: Why does this say "fistula" in parentheses?

    * - parent alive

>       - Parent simulant is still alive

Reviewer: This state name and definition will need to get updated with whatever new names we settle on.

>     * - parent dead

Reviewer: This state name should get updated to match the updated graph.
